### ISSUE TITLE
Stuck on "Additional Security Check Is Required"

### ISSUE THREAD

**Issue by d4nl** [MrKrabat/plugin.video.wakanim#33]
It seems wakanim has implemented a captcha that the plugin cannot, for obvious reasons, handle by itself.

My question is what can can be done?
- Would changing the browser id string (in api start() and getCookies() etc) to match my browser's help?

![captcha](https://user-images.githubusercontent.com/56335552/99452600-0850b300-2924-11eb-89d7-cc58910884e8.png)

**Comment by d4nl** [comment]
Let me answer my own question here. No, it makes no difference at all.

My guess is it's cookie-related.

**Comment by MrKrabat** [comment]
In the past, Wakanim's security check could be bypassed by first logging in using a web browser from the same network.
Maybe it still works?

**Comment by KyKyIIIKuH** [comment]
not work, I checked

**Comment by MrKrabat** [comment]
Unfortunately we cannot do anything against this.

Wakanim uses Incapsula to protect their server. Incapsula is executing Javascript code to prove that the request is from a real browser. This is the end of this addon :(

**Comment by KuroganeJan** [comment]
That's sad. I've came across something, but since i have no Experience with Python or JS i don't understand if it could help. So i just leave it here. https://github.com/ziplokk1/incapsula-cracker-py3

**Comment by MrKrabat** [comment]
This unofficial addon was verified by a code review from Wakanim. Wakanim allowed the existence of this unofficial addon.
I will not implement anything that circumvents Wakanim's security mechanisms.

There are options, but all would break the TOS of wakanim.

**Comment by KuroganeJan** [comment]
Understandable. Thx for the great work till now

**Comment by MrKrabat** [comment]
Right now it is working for me.

Maybe they used Incapsula only temporary for some reasons?

**Comment by Lutho08** [comment]
hi i can't log on your addons i'm French with a premium wakanim count? do you have  a fix ? thank you

**Comment by Coren** [comment]
It seems there is an api documented by wakanim here : http://account.wakanim.tv/Help 

@MrKrabat Could this be possible to use it instead of scrapping website ?

**Comment by KyKyIIIKuH** [comment]
To solve the problem, use this:

install selenium addon
https://github.com/nkvoronov/script.module.selenium

install docker addon
https://forum.libreelec.tv/thread/6721-linuxserver-io-docker-addons-and-general-docker-info/

docker pull selenium/standalone-chrome
https://hub.docker.com/r/selenium/standalone-chrome/

Checked it works.

Proof of
[https:///ploader.ru/img/_VGSFC2IUr4.jpg](Pic1)
[https:///ploader.ru/img/q7-bJ5YTP3Y.jpg](Pic2)
[https:///ploader.ru/img/u3d-YalpnP0.jpg](Pic3)

**Comment by MrKrabat** [comment]
> It seems there is an api documented by wakanim here : http://account.wakanim.tv/Help
> 
> @MrKrabat Could this be possible to use it instead of scrapping website ?

I dont think this is very usefull. This documention does not contain any informations about authentication and authorization. In my opinion this was published by accident.

I already thought of reverse engineering the wakanim api, but this will break the TOS. The windows uwp app is writen in javascript, so it is pretty easy to do.

**Comment by Lukas1818** [comment]
The Netflix Kodi plugin provide the option to [login by using  a key file](https://github.com/CastagnaIT/plugin.video.netflix/wiki/Login-with-Authentication-key) to solve the captcha problem.
May something similar is possible for Wakanim too.

**Comment by C4ET4uK** [comment]
> To solve the problem, use this:
> 
> install selenium addon
> https://github.com/nkvoronov/script.module.selenium
> 
> install docker addon
> https://forum.libreelec.tv/thread/6721-linuxserver-io-docker-addons-and-general-docker-info/
> 
> docker pull selenium/standalone-chrome
> https://hub.docker.com/r/selenium/standalone-chrome/
> 
> selenium-server-standalone
> https://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar
> https://chromedriver.chromium.org/downloads
> 
> Checked it works.
> 
> Proof of
> https://ploader.ru/img/3PZwhxq4CcE.jpg
> https://ploader.ru/img/_VGSFC2IUr4.jpg
> https://ploader.ru/img/q7-bJ5YTP3Y.jpg
> https://ploader.ru/img/u3d-YalpnP0.jpg

Can you please explain how does it work? It's just selenium for Kodi, correct?

**Comment by KyKyIIIKuH** [comment]
yes selenium, but it doesn't work anymore
https://github.com/MrKrabat/plugin.video.wakanim/issues/34

**Comment by MrKrabat** [comment]
This addon will be deleted from the Kodi addon repository soon.

Fortunately, Wakanim offers a variety of apps for different platforms. Maybe one of them can be a replacement for you

**Comment by Lukas1818** [comment]
Does anyone plan to rewrite the add-on by using the wakanim api?
